src/cljs/athens/events.cljs

```diff
@@ -552,46 +563,134 @@
 
 
 (defn indent
-  [uid]
+  "When indenting a single block:
+  - retract block from parent
+  - make block the last child of older sibling
+  - reindex parent
+  Only indent a block if it is not the zeroth block (first child).
+
+  Uses `value` to update block/string as well. Otherwise, if user changes block string and indents, the local string
+  is reset to original value, since it has not been unfocused yet (which is currently the transaction that updates the string)."
+  [uid value]
   (let [block (db/get-block [:block/uid uid])
-        parent (db/get-parent [:block/uid uid])
-        older-sib (->> parent
-                       :block/children
-                       (filter #(= (dec (:block/order block)) (:block/order %)))
-                       first
-                       :db/id
-                       db/get-block)
-        new-block {:db/id (:db/id block) :block/order (count (:block/children older-sib))}
-        reindex (dec-after (:db/id parent) (:block/order block))]
-    {:fx [[:dispatch [:transact [[:db/retract (:db/id parent) :block/children (:db/id block)]
-                                 {:db/id (:db/id older-sib) :block/children [new-block]} ;; becomes child of older sibling block — same parent but order-1
-                                 {:db/id (:db/id parent) :block/children reindex}]]]]}))
+        block-zero? (zero? (:block/order block))]
+    (when-not block-zero?
+      (let [parent (db/get-parent [:block/uid uid])
+            older-sib (->> parent
+                           :block/children
+                           (filter #(= (dec (:block/order block)) (:block/order %)))
+                           first
+                           :db/id
+                           db/get-block)
```

Comment:
There's probably a better way to get `older-sib`, or could at least abstract this into a fn.